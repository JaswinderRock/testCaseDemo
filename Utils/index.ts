export const findByTestAttr = (component: any, attr: String) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}
export const findByClassName = (component: any, className: String) => {
    const wrapper = component.find(`.${className}`);
    return wrapper;
}