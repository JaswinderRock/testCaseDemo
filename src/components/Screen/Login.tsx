import React, { Component } from 'react';
export class Login extends Component {
    render() {
        return (
            <div>
                <label>Login Here!</label>
            </div>
        )
    }
}