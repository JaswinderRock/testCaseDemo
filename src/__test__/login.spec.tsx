import React from 'react';
import { shallow } from "enzyme"
import { debug } from 'console';
import Login from '../components/Screen/Login';
import { findByClassName } from '../../Utils/index';

const setUp = () => {
    const component = shallow(<Login />);
    return component;
}
describe("Render Login", () => {
    let component: any;
    beforeEach(() => {
        component = setUp();
    });
    test("Login test is render", () => {
        const component = shallow(<Login />)
        debug(component)
        console.log(component.debug())
        expect(component).toMatchSnapshot()
    }),
        test('Should Serach for Wrapper div', () => {
            const wrapper = findByClassName(component, 'wrapper-div');
            expect(wrapper.length).toBe(1)
        }),
        test('Serach for Wrapper div', () => {
            const wrapper = findByClassName(component, 'login-form');
            expect(wrapper.length).toBe(1)
        }),
        test('Find the Label', () => {
            const wrapper = findByClassName(component, 'Login-label');
            expect(wrapper.length).toBe(1)
        }),
        test('Serach for email wrapper', () => {
            const wrapper = findByClassName(component, 'email-wrapper');
            expect(wrapper.length).toBe(1)
        }),
        test("Should Serach for Email and Password label", () => {
            const wrapper = component.find('Email *' && 'Password *');
            debug(wrapper)
        }),
        test("Should Serach for Email input", () => {
            const wrapper = component.find('.email-input');
            debug(wrapper)
        }),
        test("Email input type should by email", () => {
            const wrapper = component.find('email');
            debug(wrapper)
        }),
        test("Should Serach for password input", () => {
            const wrapper = findByClassName(component, 'password-input');
            debug(wrapper)
        }),
        test("Password input type should by password", () => {
            const wrapper = component.find('password');
            debug(wrapper)
        }),
        test(" Should Serach for Remeber wrapper ", () => {
            const wrapper = findByClassName(component, 'wrapper-checkbox');
            debug(wrapper)
        }),
        test("Should Serach for submit btn", () => {
            const wrapper = findByClassName(component, 'submit-button')
            debug(wrapper)
        })
})