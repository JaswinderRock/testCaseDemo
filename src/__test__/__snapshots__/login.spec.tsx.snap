// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Login test is render 1`] = `
<div
  className="wrapper-div"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <form
    className="login-form"
    onSubmit={[Function]}
    style={
      Object {
        "background": "#dadada",
        "border": "2px solid #0000FF ",
        "color": "blue",
        "margin": "auto",
        "marginTop": "50px",
        "padding": "40px",
      }
    }
  >
    <h3
      className="Login-label"
      id="test-id-login"
      style={
        Object {
          "fontSize": "24px",
        }
      }
    >
      Login
    </h3>
    <div
      className="email-wrapper"
    >
      <label
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Email *
      </label>
      <br />
      <input
        className="email-input"
        id="data-test-email"
        onChange={[Function]}
        placeholder="Please enter  your email"
        style={
          Object {
            "height": "3vh",
            "marginTop": "10px",
          }
        }
        type="email"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Password *
      </label>
      <br />
      <input
        className="password-input"
        onChange={[Function]}
        placeholder="Please enter your password"
        style={
          Object {
            "height": "3vh",
            "marginTop": "10px",
          }
        }
        type="password"
        value=""
      />
    </div>
    <br />
    <div
      className="wrapper-checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        Remeber me ?
      </label>
    </div>
    <br />
    <div>
      <button
        className="submit-button"
        style={
          Object {
            "background": "blue",
            "color": "white",
            "height": "5vh",
            "marginTop": "16px",
            "width": "60%",
          }
        }
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
